### PR TITLE
Feat: Added Filter

### PR DESCRIPTION
**Pull request by anamkaa**
Added filter functionality

- Filter by sort
- Filter by price
- Filter by search
- Filter by category
- Filter by stock
- Filter by rating

https://user-images.githubusercontent.com/75376715/171464375-9c8f7007-d506-4d76-95f2-36d3e969b4b5.mp4

### CODE REVIEW

**Review by hrshmistry — COMMENTED** [review]
Looking great so far! Keep up with this, btw folder structure is nicely done👍🏻

File: src/components/card/Productcard.jsx [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
   return (
-   <>
-   <div className="gh-card card card-image shadow">
+    <>
+      <div className="gh-card card card-image shadow">
         <div className="card-image-container">
           <i className="ph-heart card-icon-heart gh-heart"></i>
-          <img src={product.imgSrc} alt="product-image" className="card-image" />
+          <img
+            src={product.imgSrc}
```
Comment (Tanjum786) [inline src/components/card/Productcard.jsx:14]: you can destructure this where it repeats again and again like the **product.imgsrc product.name** etc

File: src/components/filter/Filter.jsx [hunk 2]
```diff
@@ -1,89 +1,214 @@
...
-        </div>
+      <aside className="gh-filter-wrap" id="gh-filter-wrap">
         <div className="gh-filter-wrap-row">
-          <input type="checkbox" id="air-purify"></input>
-          <label for="air-purify" className="gh-filter-text">Garden-Tools</label>
+          <div className="h6 text-bold gh-filter-title">Filters</div>
+          <div
+            className="h6 gh-filter-clear gh-filter-title"
+            id="btn-filter-clear"
+            style={{ cursor: "pointer" }}
```
Comment (Tanjum786) [inline src/components/filter/Filter.jsx]: avoid inline styling

File: src/components/navbar/Navbar.jsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { User, HeartStraight , ShoppingCartSimple ,Tree } from "phosphor-react";
 import { Link } from "react-router-dom";
 import { useCart } from '../../context/cart-context';
 import { useWishlist } from '../../context/wishlist-context';
+import { useFilter } from '../../context/filter-context';
```
Comment (Tanjum786) [inline src/components/navbar/Navbar.jsx:6]: you can try to create barrel import-export

File: src/pages/Productlisting.jsx [hunk 4]
```diff
@@ -1,36 +1,65 @@
...
+import Filter from "../components/filter/Filter";
+import Navbar from "../components/navbar/Navbar";
+import { useProduct } from "../context/product-context";
+import { useFilter } from "../context/filter-context";
+import { getSortedProduct } from "../utils/sort-by-price";
+import { getCategoryProduct } from "../utils/filter-by-category";
+import { getRatingProduct } from "../utils/filter-by-rating";
+import { getStockProduct } from "../utils/filter-by-stock";
+import { getSearchProduct } from "../utils/filter-by-search";
+import { getPricedProduct } from "../utils/filter-by-price";
```
Comment (Tanjum786) [inline src/pages/Productlisting.jsx]: here as also you can use barrel exports and imports

**Review by Tanjum786 — APPROVED** [review]
suggested some changes.

apart from that,
- all looks good  to me
- like the UI
- a good naming convention

File: src/pages/Productlisting.jsx (hunk 4 above)
Comment (anamkaa) [reply src/pages/Productlisting.jsx]: done

File: src/components/filter/Filter.jsx (hunk 2 above)
Comment (anamkaa) [reply src/components/filter/Filter.jsx]: done

**Pull request merged**